pkg/common/container.go

```diff
@@ -23,39 +23,62 @@ import (
 
 	ospdirectorv1beta1 "github.com/openstack-k8s-operators/osp-director-operator/api/v1beta1"
 	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-func getOSPVersion(parsedVersion string) (ospdirectorv1beta1.OSPVersion, error) {
+// GetOSPVersion -
+func GetOSPVersion(parsedVersion string) (ospdirectorv1beta1.OSPVersion, error) {
+	log.Log.Info(fmt.Sprintf("Parsed OSP Version %s", parsedVersion))
 	switch parsedVersion {
 	case string(ospdirectorv1beta1.TemplateVersionTrain):
-		return ospdirectorv1beta1.TemplateVersionTrain, nil
+		return ospdirectorv1beta1.TemplateVersion16_2, nil
+
+	case string(ospdirectorv1beta1.TemplateVersion16_2):
+		return ospdirectorv1beta1.TemplateVersion16_2, nil
 
 	case string(ospdirectorv1beta1.TemplateVersionWallaby):
-		return ospdirectorv1beta1.TemplateVersionWallaby, nil
+		return ospdirectorv1beta1.TemplateVersion17_0, nil
 
+	case string(ospdirectorv1beta1.TemplateVersion17_0):
+		return ospdirectorv1beta1.TemplateVersion17_0, nil
 	default:
 		err := fmt.Errorf("not a supported OSP version: %v", parsedVersion)
 		return "", err
 
 	}
 }
 
-// GetVersionFromImageURL -
-func GetVersionFromImageURL(imageURL string) (ospdirectorv1beta1.OSPVersion, error) {
+// GetOSPVersionFromImageURL -
+func GetOSPVersionFromImageURL(r ReconcilerCommon, imageURL string) (ospdirectorv1beta1.OSPVersion, error) {
 
 	parts := strings.SplitN(imageURL, ":", 2)
 	if len(parts) != 2 {
 		return "", errors.Errorf(`Invalid image name URL "%s", expected url:tag`, imageURL)
 	}
 
+	url := parts[0]
 	tag := parts[1]
 
+	// check for downstream version in image URL tags
 	// get the first major.minor version string
 	re := regexp.MustCompile(`(\d+\.\d+)?`)
 	match := re.FindStringSubmatch(tag)
 
-	// verify parsed OSP version
-	ospVersionm, err := getOSPVersion(match[1])
+	// if no match in checking image tag, check for upstream image version in URL
+	if match == nil || match[1] == "" {
+		// Until all patches backported to upstream wallaby there are two possibilities:
+		// -quay.io/tripleowallaby/openstack-tripleoclient:current-tripleo
+		// -quay.io/openstack-k8s-operators/tripleowallaby-openstack-tripleoclient:current-tripleo_patched
+		re = regexp.MustCompile(`.*\/tripleo(.*)[-\/](openstack-tripleoclient).*`)
```

Comment:
the logic in this regex strikes me as weird. Is there another way to handle this perhaps with just a flag?

Reply:
Sorry I think I do not get that comment on using a flag. This PR also adds a OpenStackRelease parameter to the ctlplane where you can always overwrite auto discovery, if that is what you meant.

Reply:
What I mean is maybe we shouldn't be using the Image URL as the mechanism to determine which version of OSP is getting used.